src/router.dispatch.ts

```diff
@@ -1,36 +1,63 @@
 import { Application, Router } from "express";
 import { HomeRouter, HealthCheckRouter, CheckDetailsRouter, ConfirmationRouter, EnterYourDetailsRouter, ConfirmCompanyRouter, CompanySearchRouter, IsBusinessRegisteredRouter, EnterBusinessNameRouter } from "./routers";
-import { errorHandler, pageNotFound } from "./routers/handlers/errors";
-import { sessionMiddleware } from "./middleware/session.middleware";
+import { csrfErrorHandler, errorHandler, pageNotFound } from "./routers/handlers/errors";
+import { COOKIE_CONFIG, sessionMiddleware } from "./middleware/session.middleware";
 import { authenticationMiddleware } from "./middleware/authentication.middleware";
 import { commonTemplateVariablesMiddleware } from "./middleware/common.variables.middleware";
 import { Urls, servicePathPrefix } from "./constants";
 import { validateUserMiddleware } from "./middleware/user.validate.middleware";
 import { featureFlagMiddleware } from "./middleware/feature.flag.middleware";
 import { localeMiddleware } from "./middleware/locale.middleware";
 import { LocalesMiddleware } from "@companieshouse/ch-node-utils";
+import helmet from "helmet";
+import { prepareCSPConfig } from "./middleware/content.policy.middleware";
+import { CsrfProtectionMiddleware } from "@companieshouse/web-security-node";
+import { EnsureSessionCookiePresentMiddleware, SessionStore } from "@companieshouse/node-session-handler";
+import { createLoggerMiddleware } from "@companieshouse/structured-logging-node";
+import { env } from "./config";
+import Redis from "ioredis";
+import { v4 as uuidv4 } from "uuid";
 
 const routerDispatch = (app: Application) => {
 
+    const nonce = uuidv4();
+
     const router = Router();
 
+    const redis = new Redis(env.CACHE_SERVER);
+    const sessionStore = new SessionStore(redis);
+    const csrfMiddlewareOptions = {
+        sessionStore,
+        enabled: true,
+        sessionCookieName: env.COOKIE_NAME
+    };
+
+    // Set nonce ids
+    app.use((req, res, next) => {
+        res.locals.cspNonce = nonce;
+        next();
+    });
```

Comment:
I think it would be better if this was it's own middleware and then imported, instead of being defined inline. 
This would mean you could add documentation to the implementation, and a descriptive name would be clearer to a reader what the intention is.